ledger/src/blockstore.rs

```diff
@@ -4049,6 +4049,13 @@ impl Blockstore {
         Ok(duplicate_slots_iterator.map(|(slot, _)| slot))
     }
 
+    pub fn has_existing_shreds_for_slot(&self, slot: Slot) -> bool {
```

Comment:
Might be nice to use this API in `next_leader_slot` (while we decide what to do there)